compiler/rustc_mir_dataflow/src/value_analysis.rs

```diff
@@ -482,27 +497,40 @@ impl<V: Clone + HasTop + HasBottom> State<V> {
     }
 
     pub fn flood_with(&mut self, place: PlaceRef<'_>, map: &Map, value: V) {
-        let StateData::Reachable(values) = &mut self.0 else { return };
-        map.for_each_aliasing_place(place, None, &mut |vi| {
-            values[vi] = value.clone();
-        });
+        self.flood_with_extra(place, None, map, value)
     }
 
-    pub fn flood(&mut self, place: PlaceRef<'_>, map: &Map) {
+    pub fn flood(&mut self, place: PlaceRef<'_>, map: &Map)
+    where
+        V: HasTop,
+    {
         self.flood_with(place, map, V::TOP)
     }
 
     pub fn flood_discr_with(&mut self, place: PlaceRef<'_>, map: &Map, value: V) {
-        let StateData::Reachable(values) = &mut self.0 else { return };
-        map.for_each_aliasing_place(place, Some(TrackElem::Discriminant), &mut |vi| {
-            values[vi] = value.clone();
-        });
+        self.flood_with_extra(place, Some(TrackElem::Discriminant), map, value)
     }
 
-    pub fn flood_discr(&mut self, place: PlaceRef<'_>, map: &Map) {
+    pub fn flood_discr(&mut self, place: PlaceRef<'_>, map: &Map)
+    where
+        V: HasTop,
+    {
         self.flood_discr_with(place, map, V::TOP)
     }
 
+    pub fn flood_with_extra(
```

Comment:
I'm not thrilled about pub method named `flood_with_extra` without any documentation. I cannot tell what the "extra" denotes when I look at the signature, at least not without looking at every other `flood` method.

After review, I infer that this is meant to be the most general form of `flood[_with]`, and every variation on `flood` should call into it (potentially indirectly) right?

(To be fair, I don't know if we have a standard convention for what to name such "core methods". )

But I'd either 1. call the pub one "flood_with_tail_elem" (since that's all that's being added from the viewpoint of the public user of the API), or 2. add some docs to the `flood_with_extra`, if only to make it clear that it may get new things added (or changed) as the internals of the value_analysis module changes.